File: week2/set-mismatch.java
// Uses the multiset class in Multiset.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        Multiset<Integer> ms = new Multiset<>();
        for (int num : nums) {
            ms.add(num);
        }
        int[] result = new int[2];
        for (int i = 1; i <= nums.length; i++) {
            if (!ms.contains(i)) {
                result[1] = i;
            } else if (ms.count(i) == 2) {
                result[0] = i;
            }
        }
        return result;
    }
}

File: summer2021/unit2/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        found = set()
        duplicated = 0
        for n in nums:
            if n in found:
                duplicated = n
            found.add(n)
                
        for i in range(1, len(nums) + 1):
            if i not in found:
                return [duplicated, i]